Marlin/src/feature/pause.cpp

```diff
@@ -252,94 +256,149 @@ void wait_for_filament_reload(const int8_t max_beep_count/*=0*/) {
-      }
-
-      // Show "Extrude More" / "Resume" menu and wait for reply
-      KEEPALIVE_STATE(PAUSED_FOR_USER);
-      wait_for_user = false;
-      lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
-      while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
-      KEEPALIVE_STATE(IN_HANDLER);
-
-      extrude_length = ADVANCED_PAUSE_EXTRUDE_LENGTH;
-
-      // Keep looping if "Extrude More" was selected
-    } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
-
+    if (swap_spool_disabled){ 
+        
+	  do {
+	    if (extrude_length > 0) {
+		  // "Wait for filament extrude"
+		  if (swap_spool_disabled) lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_EXTRUDE); 
+		  // Extrude filament to get into hotend
+          do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
+	    }
+		// Show "Extrude More" / "Resume" menu and wait for reply
+		KEEPALIVE_STATE(PAUSED_FOR_USER);
+		wait_for_user = false;
+		lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
+	    while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
+		KEEPALIVE_STATE(IN_HANDLER);
+		
+		extrude_length = PAUSE_EXTRUDE;
+		// Keep looping if "Extrude More" was selected
+	  } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
+  }
+  else do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
   #endif
-
+  
+  // Extruder purged = initialisation  
+  planner.set_e_position_mm(0); current_position[E_AXIS] = 0; destination[E_AXIS]=0;
+  
   #if ENABLED(ULTIPANEL)
     // "Wait for print to resume"
-    lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+    if (swap_spool_disabled)lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+  #endif
+	
+  // Stop blowing after delay
+  #ifdef PAUSE_DWELL
+    delay(PAUSE_DWELL);
+  #endif
+  #if (PAUSE_FANSPEED >0 && PAUSE_FAN < FAN_COUNT)
+    fanSpeeds[PAUSE_FAN]=0 ;
+  #endif  
+  
+  // Retractation before return to print
+  #ifdef PAUSE_RET
+    do_pause_e_move(-PAUSE_RET, PAUSE_RET_F);
   #endif
-
-  // Set extruder to saved position
-  planner.set_e_position_mm((current_position[E_AXIS] = resume_position[E_AXIS]));
 
   // Move XY to starting position, then Z
-  do_blocking_move_to_xy(resume_position[X_AXIS], resume_position[Y_AXIS], PAUSE_PARK_XY_FEEDRATE);
-  do_blocking_move_to_z(resume_position[Z_AXIS], PAUSE_PARK_Z_FEEDRATE);
-
-  #if ENABLED(FILAMENT_RUNOUT_SENSOR)
+  do_blocking_move_to_xy(resume_position[X_AXIS], resume_position[Y_AXIS], PAUSE_XY_F);
+
+   //Set Z_AXIS to saved position  
+  do_blocking_move_to_z(resume_position[Z_AXIS], PAUSE_Z_F);
+  
+  // Retractation Recovery
+  #ifdef PAUSE_RET
+    #if ENABLED(FWRETRACT)
+      do_pause_e_move(PAUSE_RET, RETRACT_RECOVER_FEEDRATE);    
+    #elif  PAUSE_EXTRUDE_F
+      do_pause_e_move(PAUSE_RET, PAUSE_EXTRUDE_F); 
+    #else 
+      do_pause_e_move(PAUSE_RET, 5); 
+    #endif
+  #endif
```

Comment:
Restarting must be perfect , if too quick , the extrusion line will have a hole
And Pause process don't know what are the feedrates of the printing to adapt it when recover
3 choices
Use Fwretract feedrate , that is used of the print ,  and it's perfect ' 
Use extrusion feedrate , very slow but very sure 
Use slow motion feedrate 

It's incredible but this detail is very important , we want to ensure recovery will prepare the nozzle perfectly , to avoid a blob or a hole , We must predict the exact time for the nozzle to be ready to print
Because Pause is blind , ' not when fwretract is enabled ' , we must use slow motion , the nozzle will have a long time to be ready 
May be a light strong start line of extrusion , but no HOLES